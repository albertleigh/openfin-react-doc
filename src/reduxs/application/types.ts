export interface IApplicationState {
    loading:boolean,
}
