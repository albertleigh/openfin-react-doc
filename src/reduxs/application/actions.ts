import { createAction, ActionFunctionAny, Action } from 'redux-actions';

export const APPLICATION_STARTED:string = 'APPLICATION_STARTED';
export const APPLICATION_READY:string = 'APPLICATION_READY';

export const applicationStarted:ActionFunctionAny<Action<void>> = createAction(APPLICATION_STARTED);
export const applicationReady:ActionFunctionAny<Action<void>> = createAction(APPLICATION_READY);
