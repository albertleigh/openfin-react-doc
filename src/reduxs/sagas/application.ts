import { delay } from 'redux-saga';
import { all, call, put, takeLatest } from 'redux-saga/effects';

import {
    APPLICATION_STARTED,
    applicationReady,
} from '..';

export function* handleApplicationLoading() {

    yield all([
        // you can add more sagas overhere
        call(delay,5000),
    ]);

    yield put.resolve(applicationReady());

}

export default function* () {
    yield takeLatest(APPLICATION_STARTED,handleApplicationLoading);
}