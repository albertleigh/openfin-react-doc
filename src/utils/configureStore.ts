import { applyMiddleware, createStore, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';

import rootReducer from '../reduxs';
import rootSaga from '../reduxs/sagas';

declare const window:any;

export default ()=>{

    const sagaMiddleware = createSagaMiddleware();
    const devtools = window.devToolsExtension?window.devToolsExtension():(f:any):any => (f);

    const middleware = compose(
        applyMiddleware(
            sagaMiddleware,
        ),
        devtools
    );

    const store = createStore(
        rootReducer,
        middleware,
    );

    sagaMiddleware.run(rootSaga as any);

    return store;

}
