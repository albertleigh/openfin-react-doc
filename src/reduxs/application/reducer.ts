import { handleActions, Action } from 'redux-actions';

import {
    IApplicationState,
} from './types';

import { APPLICATION_READY } from './actions';

const defaultState:Partial<IApplicationState>={
    loading:true,
};

export default handleActions({
    [APPLICATION_READY]:(state,action)=>({
        ...state,
        loading:false,
    }),

},defaultState);