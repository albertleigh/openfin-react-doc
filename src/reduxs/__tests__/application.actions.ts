import configureStore from 'redux-mock-store';
import * as Actions from '../application/actions';

const mockStore = configureStore();
const initialState = {};

const store = mockStore(initialState);

describe('Application actions',()=>{

    beforeEach(()=>{
        store.clearActions();
    });

    it('applicationStarted created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationStarted());
        expect(store.getActions()).toMatchSnapshot();
    });

    it('applicationReady created and dispatched correctly',()=>{
        store.dispatch(Actions.applicationReady());
        expect(store.getActions()).toMatchSnapshot();
    });

});
