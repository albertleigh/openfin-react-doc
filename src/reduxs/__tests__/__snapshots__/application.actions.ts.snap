// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application actions applicationReady created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_READY",
  },
]
`;

exports[`Application actions applicationStarted created and dispatched correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_STARTED",
  },
]
`;
